src/configuration.js

```diff
@@ -365,6 +362,10 @@ Configuration.prototype._checkLocalProjectId = function(cb) {
     } else {
       cb(new Error('config.projectId must be a string or number'), null);
     }
+  } else if (isString(env.GCLOUD_PROJECT) && !isEmpty(env.GCLOUD_PROJECT)) {
```

Comment:
Also, I would argue that this change is not in-scope of this PR. Better to PR separately so the division of concerns is maintained.